pkg/releases/consumer_test.go

```diff
@@ -0,0 +1,161 @@
+package releases
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"testing"
+
+	"github.com/atlassian/voyager/pkg/releases/deployinator/models"
+	"github.com/pact-foundation/pact-go/dsl"
+)
+
+/**
+This test tests the consumer interactions surrounding the resolve endpoints between
+voyager (consumer) and trebuchet (provider).
+
+To validate the pact against the current provider specs locally, use the contract testing cli found here:
+https://hello.atlassian.net/wiki/spaces/TESTA/pages/128609304/Contract+Testing+Getting+Started+Guide+for+Consumers
+ */
+func TestConsumerPact(t *testing.T) {
+
+	// Initialize the pact
+	pact := &dsl.Pact{
+		Consumer: "voyager",
+		Provider: "deployinator-trebuchet",
+		Host:     "localhost",
+		PactDir:  "generated/pacts",
+	}
+
+	defer pact.Teardown()
+
+	// Define the interactions in resolve and batchResolve separately, with the same pact
+
+	ResolveEndpointTest(pact)
+
+	BatchResolveEndpointTest(pact)
+
+	// Once interactions are verified successfully,
+	// a pact contract between the voyager/trebuchet should have now been generated. See contract under pacts/
+}
+
+func ResolveEndpointTest(pact *dsl.Pact) {
+	singleReleaseGroup := models.ResolutionResponseType{
+		Label:         "",
+		Service:       "ServiceName",
+		ReleaseGroups: map[string]map[string]interface{}{},
+	}
+
+	// Define request with dummy variables and headers set
+	var resolveTest = func() error {
+		u := fmt.Sprintf("http://%s:%d/api/v1/resolve", pact.Host, pact.Server.Port)
+		req, err := http.NewRequest("GET", u, nil)
+		q := req.URL.Query()
+		q.Add("service", "serviceName")
+		q.Add("environment", "dev")
+		q.Add("region", "us-east-1")
+		q.Add("account", "1233")
+		req.URL.RawQuery = q.Encode()
+
+		if err != nil {
+			return err
+		}
+
+		req.Header.Set("Accept", "application/json")
+
+		if _, err = http.DefaultClient.Do(req); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	// Define the expected response
+	pact.
+		AddInteraction().
+		Given("There is a release group that matches the service and location given").
+		UponReceiving("A request to get the release group").
+		WithRequest(dsl.Request{
+			Method:  "GET",
+			Path:    dsl.String("/api/v1/resolve"),
+			Headers: dsl.MapMatcher{"Accept": dsl.String("application/json")},
+			Query: dsl.MapMatcher{
+				"service":     dsl.String("serviceName"),
+				"environment": dsl.String("dev"),
+				"region":      dsl.String("us-east-1"),
+				"account":     dsl.String("1233"),
+			},
+		}).
+		WillRespondWith(dsl.Response{
+			Status:  http.StatusOK,
+			Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+			Body:    singleReleaseGroup,
+		})
+
+	// Verify the interaction
+	if err := pact.Verify(resolveTest); err != nil {
```

Comment:
Could use `require.NoError(t, err)` like everywhere else.

Reply:
This is a function that returns an error variable though..maybe leaving it for pact to handle would be better?

Reply:
Maybe I was not clear... I'm suggesting to do this:
```go
err := pact.Verify(resolveTest)
require.NoError(t, err)
```
This is almost the same to what we have here atm, but consistent with the style of other code.

Reply:
Ah okay, sorry, I see. Thanks